Presentation/Presentation.swift

```diff
@@ -161,17 +161,13 @@ public extension Presentation {
         return new
     }
 
-    /// Returns a new presentation where `callback` will be called with the value of a successful dismiss of `self`.
+    /// Returns a new presentation where `callback` will be called with the value of the result of `self`.
     func onValue<Value>(_ callback: @escaping (Value) -> ()) -> Presentation where P.Result == Future<Value> {
         let onDismiss = self.onDismiss
-        var value: Value?
-        var new = map { $0.onValue { value = $0 } }
+        var new = map { $0.onValue(callback) }
 
         new.onDismiss = { error in
```

Comment:
I believe this is not needed. It's part of `map` used at line 167.

Reply:
My latest clue is that the previous version is more correct than wrong, because we probably don't want the result of a presentation if the future completed 10 seconds after the screen was presented and for some reason the future was not cancelled.

Reply:
It's still possible to implement executing a callback only if onDismiss was not called. The only thing that will give us is `onValue` before `onDismiss` and `onValue` even if `onDismiss` is not called immediately. I guess the assumption is that onDismiss will always follow immediately as there is no point of a Presentation with result type Future if the future had completed.